concourse/pipelines/modelbase.py

```diff
@@ -69,10 +70,28 @@ def custom_init(self, raw_dict: dict):
         if raw_dict.get('output_dir', None):
             name = raw_dict['output_dir']
             self.add_output(name + '_path', name + '_path')
+        # normalise step-arguments, if given. For convenience we allow both primitive integers and strings
```

Comment:
what you mean are _scalar_ types

so state it: 'arguments' may either be a list of scalar values or a single scalar.

there is no need to do extensive type-checking: just convert everything into strings